From 6e22fd70ba46bbbba634d96f92ab54a00145936c Mon Sep 17 00:00:00 2001
From: Karthik Kambatla <kasha@apache.org>
Date: Wed, 8 Jul 2015 11:34:34 -0700
Subject: [PATCH 2328/2748] HADOOP-11878. FileContext#fixRelativePart should
 check for not null for a more informative
 exception. (Brahma Reddy Battula via kasha)

(cherry picked from commit 2e3d83f97b85c015f16f1c162b8ef0e7fee1ddf2)

Change-Id: If323707ad5985d058bb4a28e58b2da2c4170d779
---
 .../java/org/apache/hadoop/fs/FileContext.java     |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileContext.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileContext.java
index 64a5248..d15261c 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileContext.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileContext.java
@@ -60,6 +60,8 @@
 import org.apache.hadoop.util.ShutdownHookManager;
 import org.apache.htrace.core.Tracer;
 
+import com.google.common.base.Preconditions;
+
 /**
  * The FileContext class provides an interface to the application writer for
  * using the Hadoop file system.
@@ -275,6 +277,7 @@ private FileContext(final AbstractFileSystem defFs,
    * has been deliberately declared private.
    */
   Path fixRelativePart(Path p) {
+    Preconditions.checkNotNull(p, "path cannot be null");
     if (p.isUriPathAbsolute()) {
       return p;
     } else {
-- 
1.7.9.5

